core/utils.c

```diff
@@ -4643,3 +4643,23 @@ array_merge(dcontext_t *dcontext, bool intersect /* else union */,
     *dst_num = num;
 }
 
+bool
+stats_get_snapshot(dr_stats_t *drstats)
+{
+    if (!GLOBAL_STATS_ON())
+        return false;
+    CLIENT_ASSERT(drstats != NULL, "Expected non-null value for parameter drstats.");
+    /* We are at V1 of the structure, and we can't return less than the one
+     * field. We need to remove this assert when we add more fields.
+     */
+    CLIENT_ASSERT(drstats->size >= sizeof(dr_stats_t), "Invalid drstats->size value.");
+    drstats->basic_block_count = GLOBAL_STAT(num_bbs);
+    return true;
+}
+
+DR_API
+bool
+dr_get_stats(dr_stats_t *drstats)
```

Comment:
Generally the interface routine is in instrument.c rather than in the core: if you don't mind moving dr_get_stats() back to instrument.c.

Reply:
Done.